src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -33,12 +33,25 @@ public BuildEventArgsReader(BinaryReader binaryReader)
         }
 
         /// <summary>
+        /// Raised when the log reader encounters a binary blob embedded in the stream.
+        /// The arguments include the blob kind and the byte buffer with the contents.
+        /// </summary>
+        internal event Action<BinaryLogRecordKind, byte[]> OnBlobRead;
+
+        /// <summary>
         /// Reads the next log record from the binary reader. If there are no more records, returns null.
         /// </summary>
         public BuildEventArgs Read()
         {
             BinaryLogRecordKind recordKind = (BinaryLogRecordKind)ReadInt32();
 
+            while (IsBlob(recordKind))
```

Comment:
Seems to me that the BinaryLogger is writing the blobs at the end of the stream but the reader is expecting them at the beginning of the stream. Does this actually work? Am I reading something wrong?